### PR TITLE
Fix #184: Add support for the Push API

### LINKED ISSUE #184: Add support for the Push API
Add support for the Push API specified here:

https://developer.mozilla.org/en-US/docs/Web/API/Push_API

### PR DESCRIPTION
**Pull request by kapunga**
I did not include any of the following:
* Any item marked deprecated in MDN.

### CODE REVIEW

**Comment by kapunga** [comment]
Question wrt Push API methods in ServiceWorker.  Once ServiceWorkers are moved into experimental, should developers still be required to import push package implicits to access fields like `ServiceWorkerRegistration.pushManager`, or is it assumed once you are using experimental packages, you are using all of them?

**Comment by mseddon** [comment]
In general, users should enable experimental features per api, rather than as a whole.  For example, [ServiceWorkers](http://caniuse.com/#search=service%20workers) are supported by Opera, but the [Push API](http://caniuse.com/#search=push%20api) is not.

**Comment by kapunga** [comment]
Thanks a lot for the feedback!  I'll get to work tweaking things.

**Comment by kapunga** [comment]
Question:
What do we do when there is something defined to either accept or return an enum like this?
```
enum PushEncryptionKeyName {
    "p256dh",
    "auth"
};
```

**Comment by kapunga** [comment]
Question:
What do we do when there is something defined to either accept or return an enum like this?
```
enum PushEncryptionKeyName {
    "p256dh",
    "auth"
};
```
It seems you could offer additional safety by using an enum, but I couldn't find an example in the code and there doesn't seem to be any reference to such things here:
[[http://www.scala-js.org/doc/interoperability/facade-types.html]]

**Comment by sjrd** [comment]
https://github.com/scala-js/scala-js-dom/blob/87ab4d70649078f5cf9a4c0e0e22c189de86a198/src/main/scala/org/scalajs/dom/crypto/Crypto.scala#L872-L887

**Comment by kapunga** [comment]
Thanks.

File: src/main/scala/org/scalajs/dom/experimental/push/PushManager.scala [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+   * Possible values are  'prompt', 'denied', or 'granted'.
+   *
+   * @param options An object containing optional configuration parameters. It can have
+   * the following properties:
+   * - userVisibleOnly: A boolean indicating that the returned push subscription will only
+   *                    be used for messages whose effect is made visible to the user.
+   *
+   * MDN
+   */
+  def permissionState(options: PushSubscriptionOptions): Promise[PushPermissionState] = js.native
```
Comment (mseddon) [inline src/main/scala/org/scalajs/dom/experimental/push/PushManager.scala]: Optional parameter, so define `def permissionState(options: PushSubscriptionOptions = js.native): Promise[PushPermissionState] = js.native`

File: src/main/scala/org/scalajs/dom/experimental/push/PushManager.scala [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  /**
+   * The subscribe() method of the PushManager interface subscribes to a push service.
+   *
+   * It returns a Promise that resolves to a PushSubscription object containing details of a
+   * push subscription. A new push subscription is created if the current service worker
+   * does not have an existing subscription.
+   *
+   * MDN
+   */
+  def subscribe(options: PushSubscriptionOptions): Promise[PushSubscription] = js.native
```
Comment (mseddon) [inline src/main/scala/org/scalajs/dom/experimental/push/PushManager.scala]: Same here.

**Comment by mseddon** [comment]
That's all.

**Comment by kapunga** [comment]
@mseddon I was waiting for #187 to close, since I was waiting on an interface there.  Pushing it now.

**Comment by mseddon** [comment]
LGTM, thanks!

**Comment by mseddon** [comment]
LGTM, thanks! @sjrd will take it from here.

File: src/main/scala/org/scalajs/dom/experimental/push/package.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * The pushManager property of the ServiceWorkerRegistration interface returns
+     * a reference to the PushManager interface for managing push subscriptions;
+     * this includes support for subscribing, getting an active subscription, and
+     * accessing push permission status.
+     *
+     * MDN
+     */
+    var pushManager: PushManager = js.native
```
Comment (sjrd) [inline src/main/scala/org/scalajs/dom/experimental/push/package.scala]: This should probably be a `val`

**Comment by sjrd** [comment]
That's all.

**Comment by kapunga** [comment]
@mseddon @sjrd Thanks!  Just double checked it, it should be done!

**Comment by sjrd** [comment]
LGTM

**Pull request merged**